consent-ui/src/app/ais/password-input-page/password-input-page.component.ts

```diff
@@ -1,15 +1,49 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {ActivatedRoute} from "@angular/router";
+import {ConsentAuthorizationService} from "../../api";
 
 @Component({
   selector: 'consent-app-password-input-page',
   templateUrl: './password-input-page.component.html',
   styleUrls: ['./password-input-page.component.scss']
 })
 export class PasswordInputPageComponent implements OnInit {
+  passwordForm: FormGroup;
+  private authorizationSessionId: string;
+  private redirectCode: string;
 
-  constructor() { }
+  constructor(private consentAuthorizationService: ConsentAuthorizationService,
+              private activatedRoute: ActivatedRoute,
+              private formBuilder: FormBuilder) {
+  }
 
   ngOnInit() {
+    this.passwordForm = this.formBuilder.group({
+      pin: ['', Validators.required]
+    });
+
+    this.activatedRoute.queryParams.subscribe(
+      params => {
+        this.authorizationSessionId = params['authorizationSessionId'];
+        this.redirectCode = params['redirectCode'];
+      }
+    );
+  }
+
+  submit(): void {
+    this.consentAuthorizationService.embeddedUsingPOST(this.authorizationSessionId,
+      "", // TODO: clarify if this value generated by consent-ui
+      "", // TODO: clarify if this value generated by consent-ui
```

Comment:
X_XSRF_TOKEN - it is supplied by a server and is not generated here, but for now you should stub it